freqtrade/optimize/backtesting.py

```diff
@@ -275,8 +275,28 @@ def load_bt_data_detail(self) -> None:
                 if pair not in self.exchange._leverage_tiers:
                     unavailable_pairs.append(pair)
                     continue
-                self.futures_data[pair] = funding_rates_dict[pair].merge(
-                    mark_rates_dict[pair], on='date', how="inner", suffixes=["_fund", "_mark"])
+
+                if (pair in mark_rates_dict
+                        and len(funding_rates_dict[pair]) == 0
+                        and "futures_funding_rate" in self.config):
+                    mark_rates_dict[pair]["open_fund"] = self.config.get('futures_funding_rate')
+                    mark_rates_dict[pair]["close_fund"] = 0.0
+                    mark_rates_dict[pair]["high_fund"] = 0.0
+                    mark_rates_dict[pair]["low_fund"] = 0.0
+                    mark_rates_dict[pair]["volume_fund"] = 0.0
+                    mark_rates_dict[pair].rename(
+                        columns={'open': 'open_mark',
+                                 'close': 'close_mark',
+                                 'high': 'high_mark',
+                                 'low': 'low_mark',
+                                 'volume': 'volume_mark'},
+                        inplace=True)
+
+                    self.futures_data[pair] = mark_rates_dict[pair]
+                else:
+                    self.futures_data[pair] = mark_rates_dict[pair].merge(
+                        funding_rates_dict[pair], on='date',
+                        how="inner", suffixes=["_fund", "_mark"])
```

Comment:
I'm not sure this covers what you're actually after.

Assuming you're using OKX - which only provides funding rates for the last few months:
* this would work if you backtest with `timerange 20220101-20220301` (as the timerange stops before the funding data starts)
* if you backtest `20220101-20220501` - then funding rates dataframe will not be empty - but you'll have the "last few months" (i guess from April forward - so you may have 1000 mark candles, and 300 funding rate candles). that'll empty the mark rates for the period from 20220101-20220401 (it's an inner join) - and be "more wrong" than using the user-defined rate for for the period where "data is not available".

Reply:
FTX right now, but the same principle applies.

- Yes, correct, and any pairs within a longer timerange where empty funding rate df exists.
- Yes, true. I'll try to address this. Any pointers before I start?

Reply:
no - good job ;)

Reply:
no - good job ;)